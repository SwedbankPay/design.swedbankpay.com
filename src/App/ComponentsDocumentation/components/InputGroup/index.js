import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import InputGroupComponent from "@components/InputGroup";
import CodeTags from "@components/CodeTags";

const BasicExample = () => (
    <>
        <h2 id="basic-example">Basic example</h2>
        <p>
            Add a <CodeTags type="primary" code={"<span>"} /> with class <CodeTags type="secondary" code=".input-group-addon"/> to get a gray box on the left,
            right or both sides of an <CodeTags type="primary" code={"<input>"} /> element. The position of
            your <CodeTags type="secondary" code=".input-group-addon"/> depends on if you place it before or after the <CodeTags type="primary" code={"<input"} /> element.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure >
            <InputGroupComponent
                id="input-id-1"
                type="text"
                label="Name"
                placeholder="Name"
                prefixValue="Mr."
            /> {"\n\n"}
            <InputGroupComponent
                id="input-id-2"
                type="text"
                label="Email"
                placeholder="Email"
                prefixType="icon"
                prefixValue="email"
            /> {"\n\n"}
            <InputGroupComponent
                id="input-id-3"
                type="number"
                label="Amount"
                placeholder="Rounded to nearest $"
                prefixType="icon"
                prefixValue="attach_money"
                postfixValue=".00"
            /> {"\n\n"}
            <InputGroupComponent
                id="input-id-4"
                type="textarea"
                label="Notes"
                placeholder="Special requests"
                prefixType="icon"
                prefixValue="format_quote"
            />
        </ComponentPreview>
    </>
);

const CheckboxesAndRadios = () => (
    <>
        <h2 id="checkboxes-and-radios">Checkboxes and radios</h2>
        <p>Place any checkbox or radio option within an input groupâ€™s addon instead of text.</p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <div className="form-group">
                <div className="input-group">{"\n"}
                    <span className="input-group-addon">{"\n\t\t"}
                        <input type="checkbox"/>{"\n\t"}
                    </span>{"\n"}
                    <input type="text" className="form-control" />{"\n"}
                </div>
            </div>{"\n\n"}
            <div className="form-group">
                <div className="input-group">{"\n"}
                    <span className="input-group-addon">{"\n\t\t"}
                        <input type="radio"/>{"\n\t"}
                    </span>{"\n"}
                    <input type="text" className="form-control" />{"\n"}
                </div>
            </div>
        </ComponentPreview>
    </>
);

const ButtonAddons = () => (
    <>
        <h2 id="button-addons">Button addons</h2>
        <p>
            If you want <CodeTags type="primary" code={"<button>"} />s included with your input elements you can just add a <CodeTags type="primary" code={"<button>"} /> inside
            a input-group like you would anywhere else.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                type="text"
                postfixType="button"
                postfixValue="Button"
                postfixBtnColor="executive"
            /> {"\n\n"}
            <InputGroupComponent
                type="text"
                prefixType="button"
                prefixValue="Button"
                prefixBtnColor="guiding"
                postfixType="button"
                postfixValue="Button"
                postfixBtnColor="executive"
            />
        </ComponentPreview>
    </>
);

const WithSelect = () => (
    <>
        <h2 id="with-select">With select</h2>
        <p>
            Inserting text next to a <CodeTags type="primary" code={"<select>"} /> works just like any other input element.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                type="select"
                selectOptions={["Bounty hunter", "Smuggler", "Jedi"]}
                prefixValue="Profession"
            />
        </ComponentPreview>
    </>
);

const WithFeedbackIcon = () => (
    <>
        <h2 id="with-feedback">With feedback</h2>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                id="input-with-feedback-1"
                type="text"
                label="Name"
                placeholder="Name"
                prefixValue="Mr."
                feedbackIcon="cancel"
            />
        </ComponentPreview>
    </>
);

const ValidationStates = () => (
    <>
        <h2 id="validation-states">Validation states</h2>
        <p>
            You can also visualize validation on your input-groups. Just
            add <CodeTags type="secondary" code=".has-success" /> or <CodeTags type="secondary" code=".has-error" /> with
            a <CodeTags type="secondary" code=".input-group" />.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                type="text"
                validationState="error"
                placeholder="Name"
                prefixValue="Agent"
                defaultValue="Smith"
            />
        </ComponentPreview>
    </>
);

const Disabled = () => (
    <>
        <h2 id="disabled">Disabled</h2>
        <p>
            Just like for forms you can add <CodeTags type="secondary" code="disabled" /> to an input element to disable it.
            If you are going to disable an element in a <CodeTags type="secondary" code=".input-group" /> make sure to also add the <CodeTags type="secondary" code="disabled"/> attribute to the buttons.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                id="input-id-3"
                type="number"
                label="Amount"
                placeholder="Rounded to nearest $"
                disabled
                prefixType="icon"
                prefixValue="attach_money"
                postfixValue=".00"
            />
            <InputGroupComponent
                type="text"
                disabled
                prefixType="button"
                prefixValue="Button"
                prefixBtnColor="guiding"
                postfixType="button"
                postfixValue="Button"
                postfixBtnColor="executive"
            />
        </ComponentPreview>
    </>
);

const InputGroup = () => (
    <DocContainer docToc>
        <p className="lead">
            You can include additional information to specific input elements by adding <CodeTags type="primary" code={"<span>"} />s
            or <CodeTags type="primary" code={"<button>"} />s in a <CodeTags type="secondary" code=".input-group" />.
        </p>
        <BasicExample />
        <ButtonAddons />
        <WithSelect />
        <ValidationStates />
        <Disabled />
    </DocContainer>
);

export default InputGroup;

/* For testing */
export { BasicExample, CheckboxesAndRadios, ButtonAddons, WithSelect, WithFeedbackIcon, ValidationStates, Disabled };
