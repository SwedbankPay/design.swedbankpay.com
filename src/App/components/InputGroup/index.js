import React, { Fragment } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

export const Addon = ({ type, value, color, disabled }) => (
    (type === "button") ?
        <button type="button" className={`btn btn-${color}`} disabled={disabled}>
            {value}
        </button>
        : <span className="input-group-addon">{(type === "icon") ? <i className="material-icons">{value}</i> : value}</span>
);

const Feedback = ({ icon }) => (
    <span className="form-control-feedback">{"\n\t\t"}
        <i className="material-icons">{icon}</i>{"\n\t"}
    </span>
);

const InputGroup = ({
    id,
    type,
    placeholder,
    pattern,
    validate,
    required,
    defaultValue,
    autoComplete,
    disabled,
    readOnly,
    label,
    validationState,
    selectOptions,
    prefixValue,
    prefixType,
    prefixBtnColor,
    postfixValue,
    postfixType,
    postfixBtnColor,
    feedbackIcon,
    helpBlock,
    errorMessage,
    successMessage
}) => {
    const attrs = {
        type: type || null,
        className: "form-control",
        id: id || null,
        placeholder: placeholder || null,
        defaultValue: defaultValue || "",
        disabled: disabled || null,
        readOnly: readOnly || null,
        autoComplete: autoComplete || null,
        required: required || null,
        pattern: pattern ? "" : null,
        "data-validate": validate ? "" : null
    };

    const inputGrpClasses = classnames(
        "input-group",
        validationState ? `has-${validationState}` : null,
        feedbackIcon ? "has-feedback" : null
    );

    return (
        <div className={`form-group${disabled ? " disabled" : ""}`}>{"\n"}
            {label ? <label htmlFor={id}>{label}</label> : null}{label ? "\n" : null}
            {prefixValue || postfixValue || feedbackIcon ?
                <div className={inputGrpClasses}>{"\n"}
                    {prefixValue ? <Addon type={prefixType} value={prefixValue} color={prefixBtnColor} disabled={disabled} /> : null }{prefixValue ? "\n" : null}
                    {type === "textarea" ?
                        <textarea {...attrs}></textarea>
                        : type === "select" ?
                            <select className="form-control" disabled={disabled} readOnly={readOnly}>{"\n\t\t"}
                                {selectOptions.map((opt, i) => (
                                    <Fragment key={opt + i}>
                                        <option>{opt}</option>{(i !== selectOptions.length - 1) ? "\n\t\t" : ""}
                                    </Fragment>
                                ))}{"\n\t"}
                            </select>
                            :
                            <input {...attrs} />}
                    {"\n"}
                    {feedbackIcon ? <Feedback icon={feedbackIcon} /> : null} {feedbackIcon ? "\n" : null}
                    {postfixValue ? <Addon type={postfixType} value={postfixValue} color={postfixBtnColor} disabled={disabled} /> : null }{postfixValue ? "\n" : null}
                </div>
                :
                <>
                    {type === "textarea" ?
                        <textarea {...attrs}></textarea>
                        : type === "select" ?
                            <select className="form-control" disabled={disabled} readOnly={readOnly}>{"\n\t\t"}
                                {selectOptions.map((opt, i) => (
                                    <Fragment key={opt + i}>
                                        <option>{opt}</option>{(i !== selectOptions.length - 1) ? "\n\t\t" : ""}
                                    </Fragment>
                                ))}{"\n\t"}
                            </select>
                            :
                            <input {...attrs} />}
                </>
            }
            {helpBlock ? <div className="help-block" data-success={successMessage || null} data-error={errorMessage || null}>{helpBlock}</div> : null}
        </div>
    );
};

InputGroup.propTypes = {
    type: PropTypes.string.isRequired,
    id: PropTypes.string,
    placeholder: PropTypes.string,
    pattern: PropTypes.string,
    validate: PropTypes.bool,
    required: PropTypes.bool,
    defaultValue: PropTypes.string,
    autoComplete: PropTypes.string,
    disabled: PropTypes.bool,
    readOnly: PropTypes.bool,
    label: PropTypes.string,
    validationState: PropTypes.oneOf(["error", ""]),
    selectOptions: PropTypes.array,
    prefixValue: PropTypes.string,
    prefixType: PropTypes.oneOf(["button", "icon", ""]),
    prefixBtnColor: PropTypes.oneOf(["executive", "guiding"]),
    postfixValue: PropTypes.string,
    postfixType: PropTypes.oneOf(["button", "icon", ""]),
    postfixBtnColor: PropTypes.oneOf(["executive", "guiding"]),
    feedbackIcon: PropTypes.string,
    helpBlock: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.bool
    ]),
    errorMessage: PropTypes.string,
    successMessage: PropTypes.string
};

export default InputGroup;
