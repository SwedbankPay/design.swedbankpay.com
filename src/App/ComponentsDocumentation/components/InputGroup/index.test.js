import React from "react";
import { shallow } from "enzyme";

import InputGroup, { BasicExample, CheckboxesAndRadios, ButtonAddons, WithSelect, WithFeedbackIcon, ValidationStates, Disabled } from "./index";

describe("Documentation: InputGroup", () => {
    it("is defined", () => {
        expect(InputGroup).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<InputGroup />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("BasicExample", () => {
        it("is defined", () => {
            expect(BasicExample).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<BasicExample />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CheckboxesAndRadios", () => {
        it("is defined", () => {
            expect(CheckboxesAndRadios).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CheckboxesAndRadios />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ButtonAddons", () => {
        it("is defined", () => {
            expect(ButtonAddons).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ButtonAddons />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("WithSelect", () => {
        it("is defined", () => {
            expect(WithSelect).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<WithSelect />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("WithFeedbackIcon", () => {
        it("is defined", () => {
            expect(WithFeedbackIcon).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<WithFeedbackIcon />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ValidationStates", () => {
        it("is defined", () => {
            expect(ValidationStates).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ValidationStates />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Disabled", () => {
        it("is defined", () => {
            expect(Disabled).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Disabled />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
