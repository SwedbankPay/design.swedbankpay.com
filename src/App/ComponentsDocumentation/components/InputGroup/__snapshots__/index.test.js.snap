// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: InputGroup BasicExample renders 1`] = `
<Fragment>
  <h2
    id="basic-example"
  >
    Basic example
  </h2>
  <p>
    Add a 
    <CodeTags
      code="<span>"
      type="primary"
    />
     with class 
    <CodeTags
      code=".input-group-addon"
      type="secondary"
    />
     to get a gray box on the left, right or both sides of an 
    <CodeTags
      code="<input>"
      type="primary"
    />
     element. The position of your 
    <CodeTags
      code=".input-group-addon"
      type="secondary"
    />
     depends on if you place it before or after the 
    <CodeTags
      code="<input"
      type="primary"
    />
     element.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <InputGroup
      id="input-id-1"
      label="Name"
      placeholder="Name"
      prefixValue="Mr."
      type="text"
    />
     
    


    <InputGroup
      id="input-id-2"
      label="Email"
      placeholder="Email"
      prefixType="icon"
      prefixValue="email"
      type="text"
    />
     
    


    <InputGroup
      id="input-id-3"
      label="Amount"
      placeholder="Rounded to nearest $"
      postfixValue=".00"
      prefixType="icon"
      prefixValue="attach_money"
      type="number"
    />
     
    


    <InputGroup
      id="input-id-4"
      label="Notes"
      placeholder="Special requests"
      prefixType="icon"
      prefixValue="format_quote"
      type="textarea"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup ButtonAddons renders 1`] = `
<Fragment>
  <h2
    id="button-addons"
  >
    Button addons
  </h2>
  <p>
    If you want 
    <CodeTags
      code="<button>"
      type="primary"
    />
    s included with your input elements you can just add a 
    <CodeTags
      code="<button>"
      type="primary"
    />
     inside a input-group like you would anywhere else.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      postfixBtnColor="primary"
      postfixType="button"
      postfixValue="Button"
      type="text"
    />
     
    


    <InputGroup
      postfixBtnColor="primary"
      postfixType="button"
      postfixValue="Button"
      prefixBtnColor="secondary"
      prefixType="button"
      prefixValue="Button"
      type="text"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup CheckboxesAndRadios renders 1`] = `
<Fragment>
  <h2
    id="checkboxes-and-radios"
  >
    Checkboxes and radios
  </h2>
  <p>
    Place any checkbox or radio option within an input groupâ€™s addon instead of text.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <div
      className="form-group"
    >
      <div
        className="input-group"
      >
        

        <span
          className="input-group-addon"
        >
          
		
          <input
            type="checkbox"
          />
          
	
        </span>
        

        <input
          className="form-control"
          type="text"
        />
        

      </div>
    </div>
    


    <div
      className="form-group"
    >
      <div
        className="input-group"
      >
        

        <span
          className="input-group-addon"
        >
          
		
          <input
            type="radio"
          />
          
	
        </span>
        

        <input
          className="form-control"
          type="text"
        />
        

      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup Disabled renders 1`] = `
<Fragment>
  <h2
    id="disabled"
  >
    Disabled
  </h2>
  <p>
    Just like for forms you can add 
    <CodeTags
      code="disabled"
      type="secondary"
    />
     to an input element to disable it. If you are going to disable an element in a 
    <CodeTags
      code=".input-group"
      type="secondary"
    />
     make sure to also add the 
    <CodeTags
      code="disabled"
      type="secondary"
    />
     attribute to the buttons.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      disabled={true}
      id="input-id-3"
      label="Amount"
      placeholder="Rounded to nearest $"
      postfixValue=".00"
      prefixType="icon"
      prefixValue="attach_money"
      type="number"
    />
    <InputGroup
      disabled={true}
      postfixBtnColor="primary"
      postfixType="button"
      postfixValue="Button"
      prefixBtnColor="secondary"
      prefixType="button"
      prefixValue="Button"
      type="text"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup ValidationStates renders 1`] = `
<Fragment>
  <h2
    id="validation-states"
  >
    Validation states
  </h2>
  <p>
    You can also visualize validation on your input-groups. Just add 
    <CodeTags
      code=".has-success"
      type="secondary"
    />
     or 
    <CodeTags
      code=".has-error"
      type="secondary"
    />
     with a 
    <CodeTags
      code=".input-group"
      type="secondary"
    />
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      defaultValue="Smith"
      placeholder="Name"
      prefixValue="Agent"
      type="text"
      validationState="error"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup WithFeedbackIcon renders 1`] = `
<Fragment>
  <h2
    id="with-feedback"
  >
    With feedback
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      feedbackIcon="cancel"
      id="input-with-feedback-1"
      label="Name"
      placeholder="Name"
      prefixValue="Mr."
      type="text"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup WithSelect renders 1`] = `
<Fragment>
  <h2
    id="with-select"
  >
    With select
  </h2>
  <p>
    Inserting text next to a 
    <CodeTags
      code="<select>"
      type="primary"
    />
     works just like any other input element.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      prefixValue="Profession"
      selectOptions={
        Array [
          "Bounty hunter",
          "Smuggler",
          "Jedi",
        ]
      }
      type="select"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    You can include additional information to specific input elements by adding 
    <CodeTags
      code="<span>"
      type="primary"
    />
    s or 
    <CodeTags
      code="<button>"
      type="primary"
    />
    s in a 
    <CodeTags
      code=".input-group"
      type="secondary"
    />
    .
  </p>
  <BasicExample />
  <ButtonAddons />
  <WithSelect />
  <ValidationStates />
  <Disabled />
</DocContainer>
`;
